src/lib/mongoBackend/TriggeredSubscription.cpp

```diff
@@ -121,3 +128,14 @@ std::string TriggeredSubscription::toString(const std::string& delimiter)
 
   return ss.str();
 }
+
+
+
+/* ****************************************************************************
+*
+* TriggeredSubscription::stringFilterSet - 
+*/
+void TriggeredSubscription::stringFilterSet(StringFilter* _stringFilterP)
```

Comment:
That is true, this could be a problem.
Two ways of fixing this:
1. sem-protect the operation (we probably already have a semaphore tu use, one protecting the subCache
2. cloning the StringFilter instead of pointing to it. Not valid, as we'd need to sem-protect the cloning anyways ...

I'll look into subCache and see if we have a semaphore there that I can use here.